### PR TITLE
Refactor registrar into multiple files

### PR DESCRIPTION
**Pull request by stuartmcalpine**
In preparation for us doing more with entries other than creating them, I've reformatted the registrar class into one for each table, e.g., dataset, execution and dataset_alias.

As before they all follow the same format with a slight tweak. e.g., `datareg.Registrar.register_dataset()` now becomes `datareg.Registrar.dataset.create()` and `datareg.Registrar.execution.create()` etc.

This is in preparation for new functions like `datareg.Registrar.dataset.modify()` or `datareg.Registrar.dataset.delete()` etc. Placeholders for these functions have been added.

It also tidies the code a bit (mainly in prep for the new functions) so that `registrar.py` didn't become very long.

Right now no internal or external functionality has changed (other than the name of the function to call for registering datasets).

Will update docs to reflect changes in another branch.

### CODE REVIEW

**Review by JoanneBogart — CHANGES_REQUESTED** [review]
I've suggested a couple specific and simple changes, but additionally we should probably think a little more about the organization.   One possibility would be to make a base class - call it `RegistryTable` for now - which your new classes would subclass. `RegistryTable` would have base implementations for at least `modify` and `delete`. `register` is generally more complicated and different from one table to the next, so perhaps `RegistryTable.register( )` would always have to be overridden by the subclass.   To the extent that the different tables need, even partially, the same treatment, the base class would be a convenient place to put the common parts of the implementations.

File: src/dataregistry/registrar/execution.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        Parameters
+        ----------
+        parent : Registrar class
+            Contains db_connection, engine, etc
+        """
+
+        self.parent = parent
+
+    def create(
```
[the remark this answers is not in the document]
Comment (stuartmcalpine) [reply src/dataregistry/registrar/execution.py:24]: Yes I agree. I've changed all the functions to `register`

File: src/dataregistry/registrar/execution.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            for d in input_production_datasets:
+                values["register_date"] = datetime.now()
+                values["input_production_id"] = d
+                values["execution_id"] = my_id
+                add_table_row(conn, dependency_table, values, commit=False)
+
+            conn.commit()
+        return my_id
+
+    def delete(self):
```
[the remark this answers is not in the document]
Comment (stuartmcalpine) [reply src/dataregistry/registrar/execution.py:101]: I have updated the placeholders

**Comment by stuartmcalpine** [comment]
I have updated the structure as suggested. Now there is a base table class which the execution/dataset/etc tables will inherit. The base class holds the `delete` and `modify` function as they should be universal.

The `register` (renamed from `create`) functions are in each of their own subclasses.

The only slightly dirty bit is that I have to pass the execution subclass into the dataset subclass in the `Registrar` class. Thats because we need to create an execution from within the dataset registration also. But I dont see a way around it.

File: tests/end_to_end_tests/test_end_to_end.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 from dataregistry import DataRegistry
 from dataregistry.db_basic import SCHEMA_VERSION
-from dataregistry.registrar import _OWNER_TYPES
+#from dataregistry.registrar import get_owner_types
```
Comment (JoanneBogart) [inline tests/end_to_end_tests/test_end_to_end.py:7]: Can we get rid of this line altogether?

**Review by JoanneBogart — APPROVED** [review]
It looks good to me.  I made one trivial comment.

**Pull request merged**